mr/src/main/java/org/elasticsearch/hadoop/serialization/bulk/AbstractBulkFactory.java

```diff
@@ -120,6 +121,12 @@ void doWrite(Object value) {
 
                 pool.get().bytes(valueString);
             }
+            
+            else if (value instanceof Date) {
+                String valueString = (value == null ? "null": Long.toString(((Date) value).getTime()));
+                pool.get().bytes(valueString);
+            }
```

Comment:
This may fix the situation where a user tries to extract a date object for using in a parameter from the document, but I don't think it addresses the underlying problem that objects returned from the `DataFrameFieldExtractor` are not compatible with the `DataFrameValueWriter`.